xarray/core/dataset.py

```diff
@@ -319,6 +322,40 @@ def load(self):
 
         return self
 
+    def compute(self):
+        """Manually trigger loading of this dataset's data from disk or a
+        remote source into memory and return a new dataset. The original is
+        left unaltered.
+
+        Normally, it should not be necessary to call this method in user code,
+        because all xarray functions should either work on deferred data or
+        load data automatically. However, this method can be necessary when
+        working with many file objects on disk.
+        """
+        # Can't just do the below, because new.variables[k] is the
+        # same object as self.variables[k]!
+        # new = self.copy(deep=False)
```

Comment:
Let's fix `Dataset.copy()` at the source instead so this works, e.g.,
```
    def copy(self, deep=False):
        """Returns a copy of this dataset.

        If `deep=True`, a deep copy is made of each of the componentarray.
        Otherwise, a shallow copy is made, so each array in the new dataset
        is also a array in the original dataset.
        """
        variables = OrderedDict((k, v.copy(deep=deep))
                                for k, v in iteritems(self._variables))
        # skip __init__ to avoid costly validation
        return self._construct_direct(variables, self._coord_names.copy(),
                                      self._dims.copy(), self._attrs_copy())
```

I no longer have any idea why I thought it made sense to not copy variables in `Dataset.copy()`. Copying variables is cheap, and exposing details about the `Variable` object in the public API makes the `Dataset` abstraction leaky.